docs/developer-guide/extensions/ui-extensions.md

```diff
@@ -158,3 +158,64 @@ Below is an example of an extension using the flyout widget:
   );
 })(window);
 ```
+
+## Top Bar Action Menu Extensions
+
+The top bar panel is the action menu at the top of the application view where the action buttons are displayed like Details, Sync, Refresh. Argo CD allows you to add new button to the top bar action menu of an application.
+When the extension button is clicked, the custom widget will be rendered in a flyout panel.
+
+The extension should be registered using the `extensionsAPI.registerTopBarActionMenuExt` method:
+
+```typescript
+registerTopBarActionMenuExt(
+        component: TopBarActionMenuExtComponent,
+        title: string,
+        id: string,
+        flyout?: ExtensionComponent,
+        shouldDisplay: (app?: Application) => boolean = () => true,
+        iconClassName?: string,
+        isMiddle = false
+)
+```
+
+The callback function `shouldDisplay` should return true if the extension should be displayed and false otherwise:
+
+```typescript
+const shouldDisplay =  (app: Application) => {
+  return application.metadata?.labels?.['application.environmentLabelKey'] === "prd";
+```
+
+Below is an example of a simple extension with a flyout widget:
+
+```typescript
```

Comment:
Since the extensions need to be in javascript, so I think that in this doc, we want users to quickly get these samples up and running. And 'out of the box' (in this case, 'out of the docs'), they should just work.  The Application Status Panel extension sample WITHOUT flyout works fine as is.  I can copy the contents and put them in the extension*.js file in the /tmp/extensions folder, and it works nicely.  However, with the flyout, it doesn't work. This example doesn't work either (registerTopBarExtension is the old name?)

So, I think we could just provide the JS for this example. I got the following to work and there were no errors when the flyout appeared.  Feel free to update the md file with this, and rename the titles/names/etc. if you want.

```
((window) => {
    const shouldDisplay = () => {
      return true;
    };
    const flyout = () => {
      return React.createElement(
              "div",
              { style: { padding: "10px" } },
              "This is a flyout"
      );
    };
    const component = () => {
        return React.createElement(
                "div",
                {
                  onClick: () => flyout()
                },
                "Toolbar Extension Test"
        );
      };
    window.extensionsAPI.registerTopBarActionMenuExt(
            component,
            "Toolbar Extension Test",
            "Toolbar_Extension_Test",
            flyout,
            shouldDisplay,
            '',
            true
    );
  })(window);
```

Also, I got your https://github.com/ashutosh16/argocd-action-menu-extension/tree/main example to work, but I needed to tweak some things there. There's a CORS issue.

Reply:
![toolbarExtensionTest](https://github.com/user-attachments/assets/638fe064-2dba-4591-a9ed-626679013557)

Reply:
> Since the extensions need to be in javascript, I think that in this doc, we want users to quickly get these samples up and running.

good point, nice catch! some of the equivalent code blocks are annotated as TS code blocks instead of JS above, so I think it's currently inconsistent in the docs. I'd probably suggest rewriting the "Flyout widget" block in JS. The "Application Status" and "System Level" code blocks are also annotated as TS, but are actually pure JS, so I'd suggest changing the syntax highlight there too for consistency.

In other words, only function type signatures on this page would be in TS, but actual example usage is in JS

Reply:
@agilgur5  updated the docs , pls validate